app/src/main/java/com/github/se/eventradar/model/EventsOverviewViewModel.kt

```diff
@@ -60,36 +38,102 @@ class EventsOverviewViewModel(db: FirebaseFirestore = Firebase.firestore) : View
-          Log.d("EventsOverviewViewModel", "Error getting event contact: ", exception)
+        .addOnFailureListener { e ->
+          Log.d("EventsOverviewViewModel", "Error fetching user document: ", e)
         }
-
-    return contactEmail
   }
 
   companion object {
-    fun getLocalDateTime(dateTime: String): LocalDateTime {
+    fun eventFromDocument(document: DocumentSnapshot): Event? {
+      val data = document.data ?: return null
+      return Event(
+          eventName = data["name"] as? String ?: "Unknown Event",
+          eventPhoto = data["photo_url"] as? String ?: "",
+          start = getLocalDateTime(data["start"] as? String ?: "01/01/1970 00:00:00"),
+          end = getLocalDateTime(data["end"] as? String ?: "01/01/1970 00:00:00"),
+          location =
+              getLocation(
+                  data["location_name"] as? String ?: "Unknown Location",
+                  (data["location_lat"] as? Number)?.toDouble() ?: 0.0,
+                  (data["location_lng"] as? Number)?.toDouble() ?: 0.0),
+          description = data["description"] as? String ?: "",
+          ticket =
+              getEventTicket(
+                  data["ticket_name"] as? String ?: "",
+                  (data["ticket_price"] as? Number)?.toDouble() ?: 0.0,
+                  (data["ticket_quantity"] as? Number)?.toInt() ?: 0),
+          hostUserId = data["main_organiser"] as? String ?: "",
+          organiserList = getSetOfStrings(data["organisers_list"]),
+          attendeeList = getSetOfStrings(data["attendees_list"]),
+          category = getEventCategory(data["category"] as? String ?: "Not Specified"),
+          fireBaseID = document.id)
+    }
+
+    fun userFromDocument(document: DocumentSnapshot): User? {
+      val data = document.data ?: return null
+
+      return User(
+          userId = document.id,
+          age = (data["age"] as? Number)?.toInt() ?: 0,
+          email = data["email"] as? String ?: "",
+          firstName = data["firstName"] as? String ?: "",
+          lastName = data["lastName"] as? String ?: "",
+          phoneNumber = data["phoneNumber"] as? String ?: "",
+          accountStatus = data["accountStatus"] as? String ?: "Unknown",
+          eventsAttendeeList =
+              (data["eventsAttendeeList"] as? List<*>)?.filterIsInstance<String>() ?: emptyList(),
+          eventsHostList =
+              (data["eventsHostList"] as? List<*>)?.filterIsInstance<String>() ?: emptyList(),
+          profilePicUrl = data["profilePicUrl"] as? String ?: "",
+          qrCodeUrl = data["qrCodeUrl"] as? String ?: "",
+          username = data["username"] as? String ?: "Unknown User")
+    }
+
+    private fun getLocalDateTime(dateTime: String): LocalDateTime {
       val formatter = DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm:ss")
       return LocalDateTime.parse(dateTime, formatter)
     }
 
-    fun getSetOfStrings(data: Any?): Set<String> {
+    private fun getSetOfStrings(data: Any?): Set<String> {
```

Comment:
nit: this name is very cryptic. What set of strings is this getting? Where is this used?

Reply:
I didn't touch this function, but it is useful when you have a field that could either be a single string or a list of strings, and you want to normalize the output to always be a set of strings for consistency. I'll rename it "convertToStringSet".